packages/build/src/plugins_core/frameworks_api/util.ts

```diff
@@ -67,20 +66,24 @@ export const filterConfig = (
   allowedProperties: string[][],
   systemLog: SystemLogger,
 ): Record<string, unknown> =>
-  mapObject(obj, (key, value) => {
-    const keyPath = [...path, key]
+  Object.fromEntries(
+    Object.entries(obj)
+      .map(([key, value]): [string, unknown] | null => {
+        const keyPath = [...path, key]
 
-    if (!isAllowedProperty(keyPath, allowedProperties)) {
-      systemLog(`Discarding property that is not supported by the Deploy Configuration API: ${keyPath.join('.')}`)
+        if (!isAllowedProperty(keyPath, allowedProperties)) {
+          systemLog(`Discarding property that is not supported by the Deploy Configuration API: ${keyPath.join('.')}`)
 
-      return mapObjectSkip
-    }
+          return null
```

Comment:
I believe this block is actually a conflated filter and map. With this new pattern of returning nulls and then filtering them out, theoretically (since I'm not sure if it can happen in practice) this means we no longer support null values in the passed in config object.

I think my preferred here would be to change this block into distinct filter/map blocks, and keep the behaviour the same as before, so essentially:
```
Object.entries(obj)
  .filter(...keep only isAllowedProperty)
  .map(...return either [key, value] or [key, filterConfig(value, keyPath, allowedProperties, systemLog)])
```

Reply:
> this means we no longer support null values in the passed in config object.

the returned value is a key pair or `null`, rather than a key pair _containing_ null

i.e. it is `[key, value] | null`, not `[key, null]`

so we're filtering out these intermediate `null` key pairs (as in the pair as a whole is `null`)

i think i agree it should just be a filter and a map though!